dotnet/src/Connectors/Connectors.Memory.Pinecone/PineconeClient.cs

```diff
@@ -303,15 +300,13 @@ public async Task UpdateAsync(string indexName, PineconeDocument document, strin
             .InNamespace(indexNamespace)
             .Build();
 
-        (HttpResponseMessage response, string _) = await this.ExecuteHttpRequestAsync(basePath, request, cancellationToken).ConfigureAwait(false);
-
         try
         {
-            response.EnsureSuccessStatusCode();
+            await this.ExecuteHttpRequestAsync(basePath, request, cancellationToken).ConfigureAwait(false);
         }
-        catch (HttpRequestException e)
+        catch (HttpOperationException e)
         {
-            this._logger.LogWarning("Vector update for Document {0} failed. Message: {1}", document.Id, e.Message);
+            this._logger.LogWarning("Vector update for Document {Id} failed. {Message}", document.Id, e.Message);
```

Comment:
I think this is the place where exception handling and telemetry tasks overlap :) 
I noticed in some catch blocks we log as `Warning` and `Debug`. Since it's critical error, could you please update such places to log as `Error`, so telemetry will be consistent? Feel free to skip it and we can do that in scope of separate PR.

Reply:
Easy fix, done.